### PR TITLE
Ensure latest pkg.info_installed ensure latest

### PR DESCRIPTION
**Pull request by isbm**
### What does this PR do?
- Ensure that in case of multi-packages installed on the system, the latest is reported
- Refactors scattered version comparison code for `yumpkg`, `zypper` and `rpm`
### What issues does this PR fix or reference?

When there is installed multi-package, e.g. `kernel-default`, sometimes RPM does not returns properly sorted data, so it leads to accidental report of older version installed. While this is a general problem due to the fact that the structure is a deduplicated dictionary instead of list, still the latest version of the multi-package should be reported back, e.g. `pkg.info_installed` in Zypper.
### Previous Behavior

Sometimes multi-package returns older version installed.
### New Behavior

Multi-package always returns newest, latest version installed.
### Tests written?

Yes

### CODE REVIEW

File: salt/modules/rpm.py [hunk 1]
```diff
@@ -538,6 +551,49 @@ def info(*packages, **attr):
...
+        pkg_name = pkg_data.pop('name')
+        if pkg_name not in ret:
+            ret[pkg_name] = pkg_data.copy()
 
     return ret
+
+
+def version_cmp(ver1, ver2):
+    '''
+    .. versionadded:: 2015.5.4
```
Comment (terminalmage) [inline salt/modules/rpm.py]: Should be changed to 2015.8.9

**Comment by terminalmage** [comment]
I don't see a `list_installed` function in either yumpkg.py or zypper.py. Can you clarify?

**Comment by isbm** [comment]
@terminalmage `info_installed`.

**Comment by terminalmage** [comment]
OK, thanks. That's kind of what I thought, but wasn't sure.

**Pull request merged**